cedar-testing/src/integration_testing.rs

```diff
@@ -312,7 +312,13 @@ pub fn perform_integration_test_from_json_custom(
             "Unexpected validation errors in {test_name}: {:?}",
             validation_result.errors
         );
-    } else {
+    }
+    if !test.should_validate
+        && matches!(
+            test_impl.validation_comparison_mode(),
+            ValidationComparisonMode::AgreeOnAll
+        )
+    {
```

Comment:
I'd slightly prefer a full `match` here so that if we add a new `ValidationComparisonMode` this will break and we'll be reminded to consider whether this needs updating.  As is, any new mode we add implicitly skips this check, which makes me nervous